src/opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.cpp

```diff
@@ -257,17 +253,14 @@ void GuideRate::well_compute(const std::string& wgname,
             return;
         }
 
-        auto iter = this->values.find(wgname);
-        if (iter != this->values.end()) {
-            const auto& grv = iter->second->curr;
-            const auto time_diff = sim_time - grv.sim_time;
-            if (config.model().update_delay() > time_diff) {
-                return;
-            }
+        // Newly opened wells without calculated guide rates always need to update guide rates
+        // existing wells need to wait for guide rates to expire to update the guide rate.
+        if (this->values.count(wgname) > 0 && !this->time_to_update(sim_time, report_step)) {
+            return;
         }
 
         const auto guide_rate = this->eval_form(config.model(), oil_pot, gas_pot, wat_pot);
-        this->assign_grvalue(wgname, config.model(), { sim_time, guide_rate, config.model().target() });
+        this->assign_grvalue(wgname, config.model(), { guide_rate, config.model().target() });
```

Comment:
The time stamp should be updated here - without the need for any separate function.

Reply:
Then it will be backwards towards the original way from this PR.  Each guide rate value carries an individual `update_time` or `sim_time` for itself. We go through all the values to find the `earliest` `update_time`, we use that to determine whether the whole `GuideRate` expires. 

It is a little different from what I understand from the following comment. I thought the comment is suggesting that the `GuideRate` only maintain one single time stamp value. Sorry if I miss something here. 

> > This entire discussion suggests that we need to completely rethink our updating mechanism
> 
> I agree; I think we can change (and simplify!) the implementation to only store a timestamp in the shared `GuideRate` model, and some special casing for new well:
> 
> ```c++
> if (values.count(wname) == 0) 
>     compute_well(wname);
> 
> if (updsate_needed()) {
>      // Go through all wells.
>      update_timestamp();
> }
> ```
> 
> **Update:** I also suggest removing the explicit `compute()` and rather just have `get()` which will transparently determine whether a recalculation is needed or not.